lib/jit_preloader/active_record/base.rb

```diff
@@ -41,11 +41,22 @@ def has_many_aggregate(assoc, name, aggregate, field, default: 0)
             association_scope = klass.all.merge(association(assoc).scope).unscope(where: aggregate_association.foreign_key)
             association_scope = association_scope.instance_exec(&reflection.scope).reorder(nil) if reflection.scope
 
-            conditions[aggregate_association.table_name] = { aggregate_association.foreign_key => primary_ids }
+            # If the query uses an alias for the association, use that instead of the table name
+            table_alias_name = association_scope.references_values.first || aggregate_association.table_name
+            conditions[table_alias_name] = { aggregate_association.foreign_key => primary_ids }
+
+            # If the association is a STI child model, specify its type in the condition so that it
+            # doesn't include results from other child models
+            is_base_class = aggregate_association.klass.superclass.abstract_class? || aggregate_association.klass.superclass == ActiveRecord::Base
+            has_type_column = aggregate_association.klass.column_names.include?(aggregate_association.klass.inheritance_column)
+            if is_child_sti_model = !is_base_class && has_type_column
+              conditions[table_alias_name].merge!({ type: aggregate_association.klass.sti_name })
```

Comment:
We shouldn't hard code `type` here. 

`type:` => `aggregate_association.klass.inheritance_column`.